backend/analytics_server/mhq/api/teams.py

```diff
@@ -121,9 +121,24 @@ def update_team_repos(team_id: str, repos: List[RawOrgRepo]):
     query_validator = get_query_validator()
     team: Team = query_validator.team_validator(team_id)
 
-    team_repos = get_repository_service().update_team_repos(team, repos)
+    idempotency_key_raw_org_repo_map = {
+        str(repo.idempotency_key): repo for repo in repos
+    }
+
+    team_repos = get_repository_service().update_team_repos(
+        team, repos, idempotency_key_raw_org_repo_map
+    )
+
+    updated_repo_response: List[Dict[str, Any]] = []
+    for repo in team_repos:
+        adapted_org_repo = adapt_org_repo(repo)
+        raw_repo = idempotency_key_raw_org_repo_map.get(str(repo.idempotency_key))
+        if raw_repo:
+            adapted_org_repo["deployment_type"] = raw_repo.deployment_type.value
+
+        updated_repo_response.append(adapted_org_repo)
```

Comment:
@adnanhashmi09 so you are returning the same data for deployment type that you are getting from the client ? What if the data changes during the API call ?